tests/test_volume.py

```diff
@@ -120,6 +130,88 @@ def testProject(self):
             #  centered along the rotation axis for multiples of pi/2.
             self.assertTrue(np.allclose(vol_along_axis, prj_along_axis))
 
+    def testRotate(self):
+        # Create a stack of rotations to test. We will rotate by multiples of pi/2 from each axis.
+        rot_mat = np.empty((12, 3, 3), dtype=self.dtype)
+        axes = ["x", "y", "z"]
+        angles = [0, np.pi / 2, np.pi, 3 * np.pi / 2]
+        pairs = list(product(axes, angles))
+        for k in range(len(pairs)):
+            rot_mat[k] = Rotation.from_euler(pairs[k][0], pairs[k][1]).as_matrix()
+
+        # Rotate the basis volume (contains a 1 on each positive axis, zero elsewhere) by rotation matrices rot_mat.
+        # Nyquist frequencies are not set to zero for even resolution to improve error.
+        rot_vols = self.vols_3.rotate(0, rot_mat, nyquist=False)
+
+        # Create reference volumes
+        L = self.res
+        ref_vol = np.zeros((8, L, L, L))
+        index = 0
+        for i in range(2):
+            for j in range(2):
+                for k in range(2):
+                    ref_vol[index, L // 2 + (-1) ** i, L // 2, L // 2] = 1
+                    ref_vol[index, L // 2, L // 2 + (-1) ** j, L // 2] = 1
+                    ref_vol[index, L // 2, L // 2, L // 2 + (-1) ** k] = 1
+                    index += 1
+        ref_vol = Volume(ref_vol)
+
+        # Compare rotated volumes with appropriate reference volumes
+        atol = utest_tolerance(self.dtype)
+        for k in range(3):
```

Comment:
Can you add a comment for each group of comparisons?  I know you spent some time to figure it out; why repeat that again.

Reply:
Done.